### DISCUSSION TITLE
deprecated theme.DarkTheme() in Demo App

### DISCUSSION THREAD

**Discussion by dev9nine** [fyne-io/fyne#4188] [Q&A] [answered]
Demo app uses theme.DarkTheme():
[https://github.com/fyne-io/fyne/blob/master/cmd/fyne_demo/tutorials/theme.go#L38C38-L38C38](https://github.com/fyne-io/fyne/blob/master/cmd/fyne_demo/tutorials/theme.go#L38C38-L38C38)
which is deprecated. 

What would be the proper way to set the dark theme now?

i'm on Fedora (SwayWM) and have to set it manually

**Comment by Jacalz** [comment, marked as answer]
The proper way is not net set it actually. Like the deprecation note says, it is deprecated because it ignores the user's proffered theme and it should be up to the user which theme they want to use, not the developer. We have added support for following the [Dark Style Preference](https://blogs.gnome.org/alicem/2021/10/04/dark-style-preference/) which means that a desktop environment supporting that FreeDesktop protocol should result in the application having the correct theme.

In Rymdport, I use `settings.NewSettings().LoadAppearanceScreen(window)` from `"fyne.io/fyne/v2/cmd/fyne_settings/settings"` (see https://github.com/Jacalz/rymdport/blob/aaf2ff2a2d63228f5d4b7486952a4b0bf82e3f70/internal/ui/settings.go#L223C35-L223C79) to integrate the whole theme section of `fyne_settings` into my application. However, it has the downside of changing the theme for all Fyne applications on the system. For Flatpak applications it makes a lot of sense given that it is separated from the rest of the system and you thus can't install `fyne_settings` directly but it isn't a perfect solution.

**Reply by dev9nine** [reply]
thanks for quick response!
